src/main/java/io/stargate/sgv2/jsonapi/service/updater/DocumentUpdater.java

```diff
@@ -70,6 +87,29 @@ private boolean update(ObjectNode docToUpdate, boolean docInserted) {
     return modified;
   }
 
+  /**
+   * vectorize UpdateClause as needed, only when documents are found or upsert will be used by
```

Comment:
The request was not vectorize only if document is found and `$vectorize` field value is different.

Reply:
Added logic in documentUpdater,
Check if there is any $vectorize diff If there are docs found to update or doc to replace, then this is a necessary condition to proceed vectorization